lib/boxmaker.scad

```diff
@@ -0,0 +1,176 @@
+    children(4);
+  // back
+  translate([inner[0] + margin + thickness * 2,
+             inner[1] + margin + thickness * 2, 0])
+    children(5);
+}
+
+// An assembled version of the box for previewing.
+module layout_3d(box_inner, thickness) {
+  color("red") {
+    // top
+    translate([0, 0, box_inner[2]])
+      linear_extrude(height=thickness)
+        children(0);
+
+    // bottom
+    translate([0, box_inner[1], 0])
+      rotate([180, 0, 0])
+        linear_extrude(height=thickness)
+          children(1);
+  }
+
+  color("green") {
+    // left
+    translate([0, box_inner[1], 0])
+      rotate([90, 0, -90])
+        linear_extrude(height=thickness)
+          children(2);
+
+    // right
+    translate([box_inner[0], 0, 0])
+      rotate([90, 0, 90])
+        linear_extrude(height=thickness)
+          children(3);
+  }
+
+  // front
+  color("blue") {
+    translate([0, 0, 0])
+      rotate([90, 0, 0])
+        linear_extrude(height=thickness)
+          children(4);
+
+    // back
+    translate([box_inner[0], box_inner[1], 0])
+      rotate([90, 0, 180])
+        linear_extrude(height=thickness)
+          children(5);
+  }
+}
+
+module side_a_top(inner, thickness, tabs) {
+  side([inner[0], inner[1]],
+       thickness,
+       [tabs[3], tabs[4], tabs[3], tabs[4]],
+       [0, 0, 0, 0]);
+}
+
+module side_a(inner, thickness, tabs) {
+  side([inner[0], inner[1]],
+       thickness,
+       [tabs[0], tabs[1], tabs[0], tabs[1]],
+       [0, 0, 0, 0]);
+}
+
+// left/right
+module side_b(inner, thickness, tabs) {
+  side([inner[1], inner[2]],
+       thickness,
+       [tabs[4], tabs[2], tabs[1], tabs[2]],
+       [1, 0, 1, 0]);
+}
+
+// front/back
+module side_c(inner, thickness, tabs) {
+  side([inner[0], inner[2]],
+       thickness,
+       [tabs[3], tabs[2], tabs[0], tabs[2]],
+       [1, 1, 1, 1]);
+}
+
+module side(inner, thickness, tabs, polarity) {
+  SMIDGE = 0.1;
+  x = inner[0] + thickness * 2;
+  y = inner[1] + thickness * 2;
+
+  translate([-thickness, -thickness])
+    difference() {
+      square([x, y]);
+
+      // bottom
+      if (tabs[2] > 0)
+        translate([0, -SMIDGE, 0])
+          edge_cuts(x, tabs[2], thickness + SMIDGE, polarity[2]);
+
+      // top
+      if (tabs[0] > 0)
+        translate([0, y - thickness, 0])
+          edge_cuts(x, tabs[0], thickness + SMIDGE, polarity[0]);
+
+      rotate([0, 0, -90]) {
+        translate([-y, 0, 0]) {
+          // left
+          if (tabs[3] > 0)
+            translate([0, -SMIDGE, 0])
+              edge_cuts(y, tabs[3], thickness + SMIDGE, polarity[3]);
+
+          // right
+          if (tabs[1] > 0)
+            translate([0, x - thickness, 0])
+              edge_cuts(y, tabs[1], thickness + SMIDGE, polarity[1]);
+          }
+        }
+    }
+}
+
+module edge_cuts(length, finger_width, cut_depth, polarity) {
+  if (polarity == 0) {
+    outside_cuts(length, finger_width, cut_depth);
+  } else {
+    inside_cuts(length, finger_width, cut_depth);
```

Comment:
Unifying the variable names here would be a good idea too. I think "material" might be more apropos as it calls back to the laser-cut origins. If you feel strongly about cut_depth, we can change that in the library too. I'm not super committed to one or the other.